security/tests/integration/cmd/main.go

```diff
@@ -0,0 +1,276 @@
+		return "", err
+	}
+	defer resp.Body.Close()
+
+	bodyBytes, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return "", nil
+	}
+
+	return string(bodyBytes), nil
+}
+
+func runNodeAgentCertificateTests(env *integration.NodeAgentTestEnv, rootCert string, certChain string) error {
+	orgRootCert, err := readFile(rootCert)
+	if err != nil {
+		return fmt.Errorf("unable to read original root certificate: %v", rootCert)
+	}
+
+	orgCertChain, err := readFile(certChain)
+	if err != nil {
+		return fmt.Errorf("unable to read original certificate chain: %v", certChain)
+	}
+
+	nodeAgentIPAddress, err := env.GetNodeAgentIPAddress()
+	if err != nil {
+		return fmt.Errorf("external IP address of NodeAgent is not ready")
+	}
+
+	term := certValidationInterval
+	for i := 0; i < certValidateRetry; i++ {
+		if i > 0 {
+			glog.Infof("retry checking certificate update and validation in %v seconds", term)
+			time.Sleep(time.Duration(term) * time.Second)
+			term = term * 2
+		}
+
+		certPEM, err := readURI(fmt.Sprintf("http://%v:8080/cert", nodeAgentIPAddress))
+		if err != nil {
+			glog.Errorf("failed to read the certificate of NodeAgent: %v", err)
+			continue
+		}
+
+		rootPEM, err := readURI(fmt.Sprintf("http://%v:8080/root", nodeAgentIPAddress))
+		if err != nil {
+			glog.Errorf("failed to read the root certificate of NodeAgent: %v", err)
+			continue
+		}
+
+		if orgRootCert != rootPEM {
+			return fmt.Errorf("invalid root certificate was downloaded")
+		}
+
+		if orgCertChain == certPEM {
+			glog.Error("certificate chain was not updated yet")
+			continue
+		}
+
+		roots := x509.NewCertPool()
+		ok := roots.AppendCertsFromPEM([]byte(orgRootCert))
+		if !ok {
+			return fmt.Errorf("failed to parse root certificate")
+		}
+
+		block, _ := pem.Decode([]byte(certPEM))
+		if block == nil {
+			return fmt.Errorf("failed to parse certificate PEM")
+		}
+
+		cert, err := x509.ParseCertificate(block.Bytes)
+		if err != nil {
+			return fmt.Errorf("failed to parse certificate: %v", err)
+		}
+
+		opts := x509.VerifyOptions{
+			Roots: roots,
+		}
+
+		if _, err := cert.Verify(opts); err != nil {
+			return fmt.Errorf("failed to verify certificate: %v", err)
+		}
+
+		return nil
+	}
+
+	return fmt.Errorf("failed to check certificate update and validate after %v retry", certValidateRetry)
+}
+
+func main() {
+	kubeconfig := flag.String("kube-config", "", "path to kubeconfig file")
+	hub := flag.String("hub", "", "Docker hub that the Istio CA image is hosted")
+	tag := flag.String("tag", "", "Tag for Istio CA image")
+	rootCert := flag.String("root-cert", "", "Path to the original root certificate")
+	certChain := flag.String("cert-chain", "", "Path to the original workload certificate chain")
+
+	flag.Parse()
+
+	clientset, err := integration.CreateClientset(*kubeconfig)
+	if err != nil {
+		glog.Fatalf("failed to initialize K8s client: %s\n", err)
+	}
+
+	env := integration.NewSecretTestEnv("Secret creation and recovery test", clientset, *hub, *tag)
```

Comment:
This part can be handled by testEM.runtest() if you are using go test.  https://github.com/istio/istio/blob/master/tests/integration/example/tests/sample2/sample2_test.go#L63